app/src/controllers/plugins/uiExtensions/sagas.js

```diff
@@ -13,7 +13,9 @@ import {
 
 function* fetchExtensionsMetadata() {
   const plugins = yield select(pluginsSelector);
-  const uiExtensionPlugins = plugins.filter(
+  const publicPlugins = yield select(publicPluginsSelector);
+  const allPlugins = plugins.concat(publicPlugins);
+  const uiExtensionPlugins = allPlugins.filter(
```

Comment:
In this case `public` extensions will be fetched only after user login.
I suggest to call `fetchExtensionsMetadata` saga separately for public plugins as they are needed only for unauthorized user.
So the flow will look like this:
user enters app and there is no active user session -> fetch public plugins -> fetch public plugin extensions metadata.